routes/createAccount.js

```diff
@@ -0,0 +1,143 @@
+const express = require('express');
+const router = express.Router();
+const auth = require('../auth');
+const sql = require('mssql');
+
+router.post('/', function (req, res) {
+    (async () => {
+        let newAccount = await createAccount(req, res);
+
+        if (newAccount) {
+            req.session.loginMessage = "Account creation successful!"
+        }
+        res.redirect("/login")
+    })();
+});
+
+async function createAccount(req, res) {
+    if (!req.body || !req.body.firstName || !req.body.lastName || !req.body.username ||
+        !req.body.email || !req.body.password || !req.body.address || !req.body.phonenum ||
+        !req.body.city || !req.body.province || !req.body.postalCode || !req.body.country) {
+        req.session.loginMessage = "There was data missing from the account creation form. Please try again."
+        return false;
+    }
+
+    let firstName = req.body.firstName
+    let lastName = req.body.lastName
+    let username = req.body.username
+    let email = req.body.email
+    let phonenum = req.body.phonenum
+    let password = req.body.password
+    let address = req.body.address
+    let city = req.body.city
+    let province = req.body.province
+    let postalCode = req.body.postalCode
+    let country = req.body.country
+
+    await (async function () {
+        let pool = false
+        try {
+            pool = await sql.connect(dbConfig);
+            const ps = new sql.PreparedStatement(pool)
+            ps.input('userid', sql.VarChar(20))
+            await ps.prepare("SELECT userid FROM customer WHERE userid = @userid")
+            let r = await ps.execute({userid: username})
+
+            let user = r.recordset
+
+            let form = {
+                firstName: firstName,
+                lastName: lastName,
+                email: email,
+                phonenum: phonenum,
+                address: address,
+                city: city,
+                province: province,
+                postalCode: postalCode,
+                country: country,
+                password: password
+            }
+
+            let provinces = [
+                'Alberta',
+                'British Columbia',
+                'Manitoba',
+                'New Brunswick',
+                'Newfoundland and Labrador',
+                'Northwest Territories',
+                'Nova Scotia',
+                'Nunavut',
+                'Ontario',
+                'Prince Edward Island',
+                'Quebec',
+                'Saskatchewan',
+                'Yukon'
+            ]
+
+            if (user[0]) {
+                res.render('signup', {
+                    title: 'OnlyDucks Signup',
+                    form: form,
+                    provinces: provinces,
+                    error: 'This username is already taken. Please choose another.'
```

Comment:
![Screen Shot 2022-12-05 at 9 29 34 PM](https://user-images.githubusercontent.com/77898527/205824409-2ea0b1f0-279f-439c-a63e-432100bc6a0b.png)

I found a bug here. If you're logged in as 'bobby' and change your username to 'arnold' and click save, you are now logged in as 'arnold' and can see all his information. 

I know this line of code doesn't control that, but I got the idea here, so

Basically I think we shouldn't let people change their username, that would I think be the easiest fix. Or maybe at least not to a username that is already taken.